### DISCUSSION TITLE
Events importer as a first contribution

### DISCUSSION THREAD

**Discussion by gabkdlly** [pretix/pretix#2162] [Q&A] [answered]
We just started using Pretix for an [upcoming conference we are hosting](https://www.qtdevcon.com/), and are liking it enough that we are considering moving all our events to Pretix.  We have [a legacy system](https://www.kdab.com/software-services/scheduled-training/schedule/), which would have to be migrated.  It has many events which already have some orders.
I saw that there is already an [order import process](https://docs.pretix.eu/en/latest/development/api/import.html), and thought I could try my hand at building a similar "event import" process.  But, I am a bit wary, wondering whether this has not been implemented yet because it is a difficult task, and therefore perhaps inappropriate for a first contribution ?

**Comment by raphaelm** [comment, marked as answer]
Yes, this is most likely a prohibitively difficult task if you want to do it in a way that could become a core feature. Events are a data structure with lots and lots of data structures depending on them. Just importing events wouldn't be any good, you'd also need to import products, quotas, categories, questions, settings, pictures, plugin data, …

If you're planning on using the "event series" feature, a "date import" might be more realistic and possible to implement.

If I were you, for a one-off job like this I'd set up an event with your desired settings manually, and then use [the REST API's clone endpoint](https://docs.pretix.eu/en/latest/api/resources/events.html#endpoints) to copy the event for every entry from your old database.
